packages/paste-core/components/spinner/src/index.tsx

```diff
@@ -1,70 +1,42 @@
 import * as React from 'react';
-import * as PropTypes from 'prop-types';
-import {styled, keyframes, css, size as sizeFn} from '@twilio-paste/styling-library';
-import {LoadingIcon, LoadingIconProps} from '@twilio-paste/icons/esm/LoadingIcon';
-import {IconSize, isIconSizeTokenProp} from '@twilio-paste/style-props';
+import {useUID} from '@twilio-paste/uid-library';
+import {IconWrapper} from '@twilio-paste/icons/esm/helpers/IconWrapper';
+import type {IconWrapperProps} from '@twilio-paste/icons/esm/helpers/IconWrapper';
+import {useTheme} from '@twilio-paste/theme';
 
-const rotate = keyframes`
-  0% {
-    transform: rotate(0deg);
-  }
-  100% {
-    transform: rotate(359deg);
-  }
-`;
-
-interface SpinnerWrapperProps {
-  size?: IconSize;
-  show: boolean;
-}
-
-const SpinningWrapper = styled.div<SpinnerWrapperProps>(sizeFn, (props) =>
-  css({
-    display: 'inline-block',
-    opacity: props.show ? 1 : 0,
-    transition: 'opacity 250ms cubic-bezier(0.215, 0.61, 0.355, 1)',
-    animation: `${rotate} 2s linear infinite`,
-    '@media screen and (prefers-reduced-motion: reduce)': {
-      animation: 'none',
-    },
-  })
-);
+import {StyledCircleTrack, AnimatedStyledCircle, StyledSvg} from './styled';
+import {circleGeometry} from './constants';
 
-export interface SpinnerProps extends LoadingIconProps {
+export interface SpinnerProps extends IconWrapperProps {
   title: string;
-  size?: IconSize;
   delay?: number;
+  decorative: boolean;
 }
 
-const Spinner = React.forwardRef<HTMLDivElement, SpinnerProps>(
-  ({as, size, color, decorative, title, delay = 250}, ref) => {
+export const Spinner = React.forwardRef<HTMLDivElement, SpinnerProps>(
+  ({size, color = 'currentColor', title, as, display, decorative, delay = 250}, ref) => {
```

Comment:
Just curious- the previous version of spinner didn't have a `display` prop. What's the benefit from having it now?

Reply:
Technically this is a correction. We were extending the interface of LoadingIconProps, which does allow display, but not actually passing display down to the spinner anywhere. So typescript allowed it but it didn't do anything